File: 206.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int match(unsigned long long in)
{
	unsigned long long n = in;
	int i;
	for (i = 0; i < 19; i++) {
		if (i % 2 == 0) {
			if ((n % 10) != ((10 - i / 2) % 10)) {
				return 0;
			}
		}
		n /= 10;
	}
	return 1;
}

int main(void)
{
	unsigned long long base = 1020304050607080900;
	unsigned long long min = 1020304050607080900;
	unsigned long long max = 1929394959697989990;
	unsigned long long minRoot = sqrt(min);
	unsigned long long maxRoot = sqrt(max);

	unsigned long long i;
	for (i = minRoot; i < maxRoot; i += 10) {
		unsigned long long n = i*i;
		if (i % 10000000 == 0) {
			printf("%lld of %lld\n", i, maxRoot);
		}
		if (match(n)) {
			printf("solution: %lld\n", i);
			break;
		}
	}

	return 0;
}

File: 206.py
import math

base = 1020304050607080900
n = base
for i in range(0, 10):
	for j in range(0, 10):
		for k in range(0, 10):
			for l in range(0, 10):
				for m in range(0, 10):
					for n2 in range(0, 10):
						print n
						for o in range(0, 10):
							for p in range(0, 10):
								for q in range(0, 10):
									n = base + i*10**17 + j*10**15 + k*10**13 + l*10**11 + m*10**9 + n2*10**7 + o*10**5 + p*10**3 + q*10**1
									root = int(math.sqrt(n))
									if root*root == n:
										print root
